resources/providers/gcplib/inventory/provider.go

```diff
@@ -196,3 +274,112 @@ func mergeAssetContentType(assets []*assetpb.Asset) []*assetpb.Asset {
+		cache[keys.cacheKey] = ecsGcpCloud
+		extendedAssets = append(extendedAssets, &ExtendedGcpAsset{
+			Asset:       asset,
+			EcsGcpCloud: ecsGcpCloud,
+		})
+	}
+	return extendedAssets
+}
+
+type GcpAssetIDs struct {
+	orgId         string
+	projectId     string
+	parentProject string
+	parentOrg     string
+	cacheKey      string
+}
+
+func getAssetIds(asset *assetpb.Asset) GcpAssetIDs {
+	orgId := getOrganizationId(asset.Ancestors)
+	projectId := getProjectId(asset.Ancestors)
+	parentProject := fmt.Sprintf("projects/%s", projectId)
+	parentOrg := fmt.Sprintf("organizations/%s", orgId)
+	return GcpAssetIDs{
+		orgId:         orgId,
+		projectId:     projectId,
+		parentProject: parentProject,
+		parentOrg:     parentOrg,
+		cacheKey:      fmt.Sprintf("%s/%s", parentProject, parentOrg),
+	}
+}
+
+func getEcsGcpCloudData(ctx context.Context, crm *ResourceManagerWrapper, keys GcpAssetIDs) *fetching.EcsGcpCloud {
+	var orgName string
+	var projectName string
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		orgName = crm.getOrganizationDisplayName(ctx, keys.parentOrg)
+		wg.Done()
+	}()
+	wg.Add(1)
+	go func() {
+		projectName = crm.getProjectDisplayName(ctx, keys.parentProject)
+		wg.Done()
+	}()
+	wg.Wait()
+
+	return &fetching.EcsGcpCloud{
+		ProjectId:        keys.projectId,
+		ProjectName:      projectName,
+		OrganizationId:   keys.orgId,
+		OrganizationName: orgName,
+	}
+}
+
+func getOrganizationId(ancestors []string) string {
+	last := ancestors[len(ancestors)-1]
+	return strings.Split(last, "/")[1]
+}
+
+func getProjectId(ancestors []string) string {
+	return strings.Split(ancestors[0], "/")[1]
+}
+
+func getProjectLogMetricAssets(projectAssets []*ExtendedGcpAsset) []*ExtendedGcpAsset {
+	return lo.Filter(projectAssets, func(asset *ExtendedGcpAsset, _ int) bool {
+		return asset.AssetType == "logging.googleapis.com/LogMetric"
+	})
+}
+
+func getProjectAlertPolicyAssets(projectAssets []*ExtendedGcpAsset) []*ExtendedGcpAsset {
+	return lo.Filter(projectAssets, func(asset *ExtendedGcpAsset, _ int) bool {
+		return asset.AssetType == "monitoring.googleapis.com/AlertPolicy"
+	})
+}
+
+func getProjectAssetsMetadata(projectAssets []*ExtendedGcpAsset) (string, string, string, error) {
+	projectNames := lo.FindUniquesBy(projectAssets, func(asset *ExtendedGcpAsset) string { return asset.ProjectName })
+	organizationIds := lo.FindUniquesBy(projectAssets, func(asset *ExtendedGcpAsset) string { return asset.OrganizationId })
+	organizationNames := lo.FindUniquesBy(projectAssets, func(asset *ExtendedGcpAsset) string { return asset.OrganizationName })
+
+	if len(projectNames) != 1 || len(organizationIds) != 1 || len(organizationNames) != 1 {
+		return "", "", "", errors.New("failed to get project/organization name")
+	}
+
+	// values in index 0 are the same for all indices.
+	// verified by the lo.FindUniquesBy() above
+	return projectNames[0].ProjectName,
+		organizationIds[0].OrganizationId,
+		organizationNames[0].OrganizationName,
+		nil
+}
```

Comment:
this is a bit excessive, and was done to be extra-safe, but we can alternately just take the data from the first element, it's already guaranteed to be the same for all values, regardless of the `FindUniquesBy`, because we've grouped assets by project